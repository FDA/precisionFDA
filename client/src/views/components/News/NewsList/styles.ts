import styled from 'styled-components'
import { commonStyles } from '../../../../styles/commonStyles'


export const StyledNewsListContainer = styled.div`
  .news-list {
    ${commonStyles.listContainer}
  }

  .pfda-pagination {
    ${commonStyles.listPagination}
  }
`
