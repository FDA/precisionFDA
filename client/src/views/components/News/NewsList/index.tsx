import React, { FunctionComponent } from 'react'
import { connect } from 'react-redux'

import history from '../../../../utils/history'
import { NewsListItem, NewsListItemLarge } from '../NewsListItem'
import { INewsItem } from '../../../../types/newsItem'
import { IPagination } from '../../../../types/pagination'
import Pagination from '../../TableComponents/Pagination'
import Loader from '../../Loader'
import {
  fetchNews,
  newsListSetPage,
} from '../../../../actions/news'
import {
  newsListItemsSelector,
  newsListIsFetchingSelector,
  newsListPaginationSelector
} from '../../../../reducers/news/list/selectors'
import { StyledNewsListContainer } from './styles'


interface INewsListProps {
  listItemComponent: typeof NewsListItem,
  newsItems: INewsItem[],
  isFetching: boolean,
  filter?: (item: INewsItem[]) => INewsItem[],
  allowPagination?: boolean,
  pagination?: IPagination,
  setPageHandler?: (page: number) => void,
}


const NewsList: FunctionComponent<INewsListProps> = ({ newsItems, isFetching, listItemComponent=NewsListItemLarge, filter, allowPagination=true, pagination, setPageHandler }: INewsListProps) => {

  if (isFetching) {
    return (
      <div className='text-center'>
        <Loader />
      </div>
    )
  }

  if (newsItems.length) {
    let itemsToShow = newsItems
    if (filter) {
      itemsToShow = filter(newsItems)
    }
    const ListItem = listItemComponent

    return (
      <StyledNewsListContainer>
        <ul className="news-list">
          {itemsToShow.map((newsItem) => <ListItem key={newsItem.id} newsItem={newsItem} />)}
        </ul>
        {allowPagination ??
          <Pagination data={pagination} setPageHandler={setPageHandler} />
        }
      </StyledNewsListContainer>
    )
  }

  return <div className='text-center'>No news found.</div>
}

NewsList.defaultProps = {
  newsItems: [],
  listItemComponent: NewsListItemLarge,
  isFetching: false,
}

const mapStateToProps = (state: any) => ({
  newsItems: newsListItemsSelector(state),
  isFetching: newsListIsFetchingSelector(state),
  pagination: newsListPaginationSelector(state),
})

const mapDispatchToProps = (dispatch: any) => ({
  openNewsItem: (id: number) => history.push(`/news/${id}`),
  setPageHandler: (page: number) => {
    dispatch(newsListSetPage(page))
    dispatch(fetchNews())
  },
})

export {
  NewsList
}

export default connect(mapStateToProps, mapDispatchToProps)(NewsList)
