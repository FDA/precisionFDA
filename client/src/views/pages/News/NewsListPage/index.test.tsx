import React from 'react'
import { shallow } from 'enzyme'

import { NewsListPage } from '.'

import NewsList from '../../../components/News/NewsList'


describe('NewsListPage test', () => {
  it('should render', () => {
    const wrapper = shallow(<NewsListPage />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain NewsList', () => {
    const wrapper = shallow(<NewsListPage />)
    // console.log(wrapper.debug());
    expect(wrapper.find(NewsList)).toHaveLength(1)
  })
})
