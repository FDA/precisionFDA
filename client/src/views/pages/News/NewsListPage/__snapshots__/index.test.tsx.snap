// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsListPage test should render 1`] = `
<PublicLayout>
  <NavigationBar
    subtitle="Announcements and press for precisionFDA."
    title="News"
  />
  <div
    className="challenges-page-layout"
  >
    <div
      className="left-column"
    >
      <Connect(NewsList)
        listItemComponent={[Function]}
      />
    </div>
    <div
      className="right-column right-column--override pfda-main-content-sidebar"
    >
      <SectionHeading>
        NEWS BACKLOG
      </SectionHeading>
      <NewsYearList
        elementName="news"
        query={[Function]}
        setYearHandler={[Function]}
      />
      <hr />
      <SectionHeading>
        COMMUNITY NEWS
      </SectionHeading>
      <p>
        If you have newsworthy updates of value to the precisionFDA community, let us know!
      </p>
    </div>
  </div>
</PublicLayout>
`;
